modules/launch-templates/main.tf

```diff
@@ -0,0 +1,69 @@
+resource "aws_launch_template" "this" {
+  for_each = local.launch_template_config
+
+  name        = format("%s-%s", each.value.launch_template_prefix, var.eks_cluster_id)
+  description = "Launch Template for Amazon EKS Worker Nodes"
+
+  image_id               = each.value.ami
+  update_default_version = true
+
+  user_data = base64encode(templatefile("${path.module}/templates/userdata-${each.value.launch_template_os}.tpl",
+    {
+      pre_userdata         = each.value.pre_userdata
+      post_userdata        = each.value.post_userdata
+      bootstrap_extra_args = each.value.bootstrap_extra_args
+      kubelet_extra_args   = each.value.kubelet_extra_args
+      eks_cluster_id       = var.eks_cluster_id
+      cluster_ca_base64    = data.aws_eks_cluster.eks.certificate_authority[0].data
+      cluster_endpoint     = data.aws_eks_cluster.eks.endpoint
+  }))
+
+  iam_instance_profile {
+    name = each.value.iam_instance_profile
+  }
+
+  ebs_optimized = true
+
+  dynamic "block_device_mappings" {
+    for_each = each.value.block_device_mappings
+
+    content {
+      device_name = try(block_device_mappings.value.device_name, null)
+
+      ebs {
+        delete_on_termination = try(block_device_mappings.value.delete_on_termination, true)
+        encrypted             = try(block_device_mappings.value.encrypted, true)
+        kms_key_id            = try(block_device_mappings.value.kms_key_id, null)
+        volume_size           = try(block_device_mappings.value.volume_size, null)
+        volume_type           = try(block_device_mappings.value.volume_type, null)
+        iops                  = block_device_mappings.value.volume_type == "gp3" || block_device_mappings.value.volume_type == "io1" || block_device_mappings.value.volume_type == "io2" ? block_device_mappings.value.iops : null
+        throughput            = block_device_mappings.value.volume_type == "gp3" ? block_device_mappings.value.throughput : null
```

Comment:
If the user passes only gp3 as volume type, the module raises an error because throughput should be 125 as default value instead of 0.

Suggestion of code:
```hcl
block_device_mappings.value.volume_type == "gp3" ? try(tonumber(block_device_mappings.value.throughput)) < 125 ? 125 : block_device_mappings.value.throughput : null
```

Reply:
changed this to 125 if its gp3